File: Leetcode/0067 Add Binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        int carry = 0;
        int i = a.size() - 1;
        int j = b.size() - 1;
        string sum = "";
        
        while(i >= 0 || j >= 0){
            int ai = i < 0? 0 : a[i--] - '0';
            int bj = j < 0? 0 : b[j--] - '0';
            
            int sm = ai + bj + carry;
            sum    = to_string(sm & 1) + sum;
            carry  = sm >> 1;
        }
        
        if(carry)
            sum = to_string(1) + sum;
        
        return sum;
    }
};

File: Leetcode/0067 Add Binary.py
from itertools import zip_longest


class Solution:
    def addBinary(self, a, b):
        sum, carry = [], 0

        for ai, bi in zip_longest(reversed(a), reversed(b), fillvalue='0'):
            digit_sum = carry + int(ai) + int(bi)
            sum.append(f'{digit_sum % 2}')
            carry = digit_sum // 2

        if carry > 0:
            sum.append(f'{carry}')

        return "".join(reversed(sum))
